### ISSUE TITLE
"Missing MXSUB variable" error for conduction NetCDF file

### ISSUE THREAD

**Issue by erikgrinaker** [boutproject/BOUT-dev#100]
Running `examples/conduction/runtest` gives `ValueError: Missing MXSUB variable`. The NetCDF file has `MXSUB` as an attribute, not a variable:

```
$ h5dump data/BOUT.dmp.0.nc | grep -A 6 MXSUB
   ATTRIBUTE "MXSUB" {
      DATATYPE  H5T_STD_I32LE
      DATASPACE  SIMPLE { ( 1 ) / ( 1 ) }
      DATA {
      (0): 1
      }
   }
```

I fixed it locally by modifying `DataFile_netCDF` to handle attributes as if they were variables, but I am not sure if that's the best way to solve this, or if it has any unintended side effects.

This is with NetCDF 4.3.3.1 and python-netcdf 1.1.9. It is probably caused by BOUT not being fully compatible with NetCDF 4.3, but I'd like to try to fix that rather than downgrading if possible.

**Comment by yegakjh** [comment]
I ran into the same problem. 

I simply commented out some lines in ncxx4.cxx

there are some lines in Ncxx4::write(int) that treat variables of a scalar as "attribute". 
 
Juhyung

_________________________
diff --git a/src/fileio/impls/netcdf4/ncxx4.cxx b/src/fileio/impls/netcdf4/ncxx4.cxx
index 82d6371..884e3a3 100644
--- a/src/fileio/impls/netcdf4/ncxx4.cxx
+++ b/src/fileio/impls/netcdf4/ncxx4.cxx
@@ -466,11 +466,12 @@ bool Ncxx4::write(int *data, const char *name, int lx, int ly, int lz) {
   msg_stack.push("Ncxx4::write(int)");
 #endif

-  if(nd == 0) {
-    // Zero-dimensional attribute
+  // if(nd == 0) {
+  //   // Zero-dimensional attribute

-    NcGroupAtt att  = dataFile->putAtt(name, ncInt, 1, data);
-  }else {
+  //   NcGroupAtt att  = dataFile->putAtt(name, ncInt, 1, data);
+
+  // }else {
     // Multi-dimensional variables

     NcVar var = dataFile->getVar(name);
@@ -497,7 +498,7 @@ bool Ncxx4::write(int *data, const char *name, int lx, int ly, int lz) {
     counts[0] = lx; counts[1] = ly; counts[2] = lz;

     var.putVar(start, counts, data);
-  }
+    //  }

 #ifdef NCDF_VERBOSE
   output.write("Ncxx4:: write { Done } \n");

**Comment by ZedThree** [comment]
Fixed in #219
